File: BAEKJOON/string/10809.py
# 10809: 알파벳 찾기/브론즈 2

alphabets = {}
for i in 'abcdefghijklmnopqrstuvwxyz':
    alphabets[i] = '-1'

idx = 0
for i in input():
    if alphabets[i] != '-1':
        idx += 1
        continue
    alphabets[i] = str(idx)
    idx += 1

print(' '.join(alphabets.values()))

File: BAEKJOON/string/javascript/10809.js
const fs = require("fs");
let input =
  process.platform === "linux"
    ? fs.readFileSync("/dev/stdin").toString().trim()
    : fs
        .readFileSync(__dirname + "/input.txt")
        .toString()
        .trim();

function solution(input) {
  const result = Array(26).fill(-1);

  for (let char of input) {
    result[char.charCodeAt() % 97] = input.indexOf(char);
  }

  return result.join(" ");
}

console.log(solution(input));
